### PR TITLE
Updates to get user_phpbb3 working with OwnCloud 7

### PR DESCRIPTION
**Pull request by rabbitt**
- Fixed problems with password verification (imported password checking functionality from phpbb3)
- Cleaned up OC_User_Phpbb3 a bit, making it work with OwnCloud 7. Realistically, the class should be
  broken up a bit more but, for now it works and that is good enough. :-)
- Todo: import groups too
- created a README.md

### CODE REVIEW

**Comment by rabbitt** [comment]
Any chance you could take a look at this?

**Comment by QuingKhaos** [comment]
Wohaa, didn't saw this PR :+1: Looks great, must review it at the weekend :)

File: lib/helper.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+*
+* @package phpBB3
```
Comment (QuingKhaos) [inline lib/helper.php:4]: Is this helper copied from phpBB?

File: lib/helper.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+    * We're kind of forced to use MD5 here since it's the only
+    * cryptographic primitive available in all versions of PHP
+    * currently in use.  To implement our own low-level crypto
+    * in PHP would result in much worse performance and
+    * consequently in lower iteration counts and hashes that are
+    * quicker to crack (by non-PHP code).
+    */
+    if (PHP_VERSION >= 5) {
```
Comment (QuingKhaos) [inline lib/helper.php:86]: Uff, I don't mind supporting php version below 5.4 ;) Adequate usage of the real `crypt()` would be good.

File: lib/helper.php (hunk 1 above)
Comment (rabbitt) [inline lib/helper.php:4]: Almost a verbatim copy. I moved from procedural driven to a static class, tweaked the names (removed the phpbb_ prefix mostly), and cleaned up the code a little bit. I didn't really touch the logic however as I didn't feel like trying to parse through what they were doing .. and I wanted to make sure there was feature parity with password hashing. I figure the code can be more optimized at a later date - for now it's working. ;-)

File: lib/helper.php (hunk 2 above)
Comment (rabbitt) [inline lib/helper.php:86]: As mentioned in my other comment, the logic is pretty much a verbatim copy of the original (from phpbb). I haven't modified it and don't think it should be - at least not right now.

**Comment by QuingKhaos** [comment]
Yes, re-reading the code a little bit, we can't swap the crypt function as it wouldn't work then.

**Pull request merged**